src/jobs/ega/services/permissions.ts

```diff
@@ -126,40 +178,76 @@ export const processPermissionsForApprovedUsers = async (
         }
         break;
       case 'SERVER_ERROR':
-        logger.info(`Error fetching existing permission: ${existingPermission.message}`);
-        break;
       default:
-        logger.error(`Unexpected error fetching existing permission: ${existingPermission}`);
+        logger.error(`Error fetching existing permission: ${existingPermission.message}`);
+        totalCreatedPermissionsResult.errors.push({
+          processName: 'getPermissionsByUserId',
+          message: existingPermission.message,
+          status: existingPermission.status,
+        });
     }
 
     if (permissionRequests.length) {
       const chunkedPermissionRequests = chunk(permissionRequests, EGA_MAX_REQUEST_SIZE);
       for await (const requests of chunkedPermissionRequests) {
-        await createRequiredPermissions(egaClient, approvedUser, requests);
+        const createdPermissions = await createRequiredPermissions(
+          egaClient,
+          approvedUser,
+          requests,
+        );
+        if (createdPermissions.num_granted !== 0) {
+          userPermissionResult.permissionsGrantedCount =
+            userPermissionResult.permissionsGrantedCount + createdPermissions.num_granted;
+          totalCreatedPermissionsResult.numUsersWithNewPermissions++;
+        }
       }
     }
+
+    if (
+      userPermissionResult.permissionsGrantedCount === userPermissionResult.permissionsMissingCount
+    ) {
+      totalCreatedPermissionsResult.numUsersSuccessfullyProcessed++;
+    }
   }
   logger.info('Completed processing permissions for all DACO approved users.');
+  const endTime = new Date();
+  const timeElapsed = moment(endTime).diff(startTime, 'minutes');
+  return {
```

Comment:
My report gathering here and in `revokePermissions` feels messy, I am open to suggestions. My main goal in the results data is to track whether the expected outcomes match the actual results, i.e.
- the total number of EgaUsers inputted to `processPermissionsForApprovedUsers` matches the total number of users successfully processed
- the total number of Datasets inputted to `removeExpiredPermissions` matches the total number of datasets successfully processed

Reply:
This may be because you have 3 outcomes but only 2 statuses:
1. errors during job
2. complete with incorrect number of processed users
3. complete with correct numbers

But you are only reporting: success/failure.